internal/utils/main.go

```diff
@@ -286,6 +284,138 @@ func AddExportTypeFlags(flags *pflag.FlagSet) {
 	flags.BoolP("export-ttl", "", false, "set in order to export ttl changes")
 }
 
+// TODO: https://stellarorg.atlassian.net/browse/HUBBLE-386 better flags/params
+// Some flags should be named better
+type FlagValues struct {
+	StartNum       uint32
+	EndNum         uint32
+	StrictExport   bool
+	IsTest         bool
+	IsFuture       bool
+	Extra          map[string]string
+	UseCaptiveCore bool
+	DatastorePath  string
+	BufferSize     uint32
+	NumWorkers     uint32
+	RetryLimit     uint32
+	RetryWait      uint32
+	Path           string
+	Limit          int64
+	Bucket         string
+	Credentials    string
+	Provider       string
+}
+
+// MustFlags gets the values of the the flags for all commands.
+// If any do not exist, it stops the program fatally using the logger
+// TODO: https://stellarorg.atlassian.net/browse/HUBBLE-386 Not sure if all these arg checks are necessary
+func MustFlags(flags *pflag.FlagSet, logger *EtlLogger) FlagValues {
+	endNum, err := flags.GetUint32("end-ledger")
```

Comment:
Did we remove all functionality in stellar-etl to be able to run in an unbounded mode?

Reply:
Yes in this new section (so unbounded mode is only removed for contract_events).

I think if we want unbounded mode we should do some redesign and refactoring

Reply:
Agreed